workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/monitoring/repository/ActivityCmdaApiQueryRepository.java

```diff
@@ -31,11 +35,34 @@ public ActivityCmdaApiQueryRepository(RepositoryService repositoryService,
    * and eventually the one activity being executed (with status ONGOING).
    */
   @Override
-  public List<ActivityInstanceDomain> findAllByWorkflowInstanceId(String instanceId) {
-    List<ActivityInstanceDomain> result =
-        objectConverter.convertCollection(historyService.createHistoricActivityInstanceQuery()
-            .processInstanceId(instanceId).orderByHistoricActivityInstanceStartTime().asc()
-            .list(), ActivityInstanceDomain.class);
+  public List<ActivityInstanceDomain> findAllByWorkflowInstanceId(String workflowId, String instanceId,
+      WorkflowInstLifeCycleFilter lifeCycleFilter) {
+    HistoricActivityInstanceQuery historicActivityInstanceQuery = historyService.createHistoricActivityInstanceQuery()
+        .processInstanceId(instanceId);
+
+    if (StringUtils.isNotBlank(lifeCycleFilter.getStartedBefore())) {
+      historicActivityInstanceQuery.startedBefore(new DateTime(lifeCycleFilter.getStartedBefore()).toDate());
+    }
+
+    if (StringUtils.isNotBlank(lifeCycleFilter.getStartedAfter())) {
+      historicActivityInstanceQuery.startedAfter(new DateTime(lifeCycleFilter.getStartedAfter()).toDate());
+    }
+
+    if (StringUtils.isNotBlank(lifeCycleFilter.getFinishedBefore())) {
+      historicActivityInstanceQuery.finishedBefore(new DateTime(lifeCycleFilter.getFinishedBefore()).toDate());
+    }
+
+    if (StringUtils.isNotBlank(lifeCycleFilter.getFinishedAfter())) {
+      historicActivityInstanceQuery.finishedAfter(new DateTime(lifeCycleFilter.getFinishedAfter()).toDate());
+    }
+
+    List<ActivityInstanceDomain> result = objectConverter.convertCollection(historicActivityInstanceQuery
+        .orderByHistoricActivityInstanceStartTime()
+        .asc()
+        .list()
+        .stream()
+        .filter(actInstance -> workflowId.equals(actInstance.getProcessDefinitionKey()))
```

Comment:
the check is already done at service level, i think this filter is not necessary, right?